db/version_set.cc

```diff
@@ -2492,6 +2492,7 @@ Status VersionSet::LogAndApply(ColumnFamilyData* column_family_data,
   // because &w is ensuring that all new writes get queued.
   {
 
+    EnvOptions opt_env_opts = env_->OptimizeForManifestWrite(env_options_);
```

Comment:
I think `GetTableProperties`, `GetMemoryUsageByTableReaders`, `ApproximateSize` also access it outside the lock. Maybe store a copy in `Version` is a general solution to those cases.